src/promptflow-devkit/promptflow/_sdk/operations/_flow_operations.py

```diff
@@ -997,102 +999,123 @@ def _merge_signature(extracted, signature_overrides):
                     f"Ports from entry: {', '.join(signature[key].keys())}\n"
                 )
 
+            # TODO: merge the signature
             signature[key] = signature_overrides[key]
 
         return signature
 
     @monitor_operation(activity_name="pf.flows._infer_signature", activity_type=ActivityType.INTERNALCALL)
-    def _infer_signature(self, entry: Callable, keep_entry: bool = False, validate: bool = True) -> Tuple[dict, Path]:
+    def _infer_signature(
```

Comment:
flow save has its own activity monitor, why add additional monitor for this internal call?

Reply:
May save our telemetry cost by removing this